// megalo-router-patch/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _methods = require('./utils/methods');

var _query = require('./utils/query');

var _Vue = void 0;

var routerPatch = {
    install: function install(Vue, options) {

        if (this.installed && _Vue === Vue) return;
        this.installed = true;
        _Vue = Vue;

        var _route = {};
        var _router = {};

        var handleMixin = function handleMixin(that) {
            if (that.$parent) return;
            var $mp = that.$root.$mp;

            _route = (0, _query.parseRoute)($mp);
            _router = {
                app: that,
                mode: 'history',
                currentRoute: _route,
                push: function push(location, complete, fail, success) {
                    return (0, _methods.push)(location, complete, fail, success, $mp.platform);
                },
                replace: function replace(location, complete, fail, success) {
                    return (0, _methods.replace)(location, complete, fail, success, $mp.platform);
                },
                go: function go(delta) {
                    return (0, _methods.go)(delta, $mp.platform);
                },
                back: function back() {
                    return (0, _methods.back)($mp.platform);
                }
            };
        };

        Vue.mixin({
            onShow: function onShow() {
                handleMixin(this);
            },
            beforeCreate: function beforeCreate() {
                handleMixin(this);
            }
        });

        Object.defineProperty(Vue.prototype, '$router', {
            get: function get() {
                return _router;
            }
        });

        Object.defineProperty(Vue.prototype, '$route', {
            get: function get() {
                return _route;
            }
        });
    }
};

exports.default = routerPatch;
