// megalo-router-patch/dist/utils/query.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
var stringifyQuery = function stringifyQuery(obj) {
    var res = obj ? Object.keys(obj).map(function (key) {
        var val = obj[key];
        if (val === undefined) {
            return '';
        }
        if (val === null) {
            return key;
        }
        if (Array.isArray(val)) {
            var result = [];
            val.forEach(function (val2) {
                if (val2 === undefined) {
                    return;
                }
                if (val2 === null) {
                    result.push(key);
                } else {
                    result.push(key + '=' + val2);
                }
            });
            return result.join('&');
        }
        return key + '=' + val;
    }).filter(function (x) {
        return x.length > 0;
    }).join('&') : null;
    return res ? '?' + res : '';
};

var parseUrl = function parseUrl(location) {
    if (typeof location === 'string') return location;
    var path = location.path,
        query = location.query;

    var queryStr = stringifyQuery(query);
    return '' + path + queryStr;
};

var parseRoute = function parseRoute($mp) {
    var _$mp = $mp || {};
    var path = _$mp.page && _$mp.page.route;
    return {
        path: '/' + path,
        query: _$mp.query,
        fullPath: parseUrl({
            path: '/' + path,
            query: _$mp.query
        }),
        name: path && path.replace(/\/(\w)/g, function ($0, $1) {
            return $1.toUpperCase();
        })
    };
};

exports.parseUrl = parseUrl;
exports.parseRoute = parseRoute;
